### DISCUSSION TITLE
what is the maximum limit for sending video, audio, and pdf in MD?

### DISCUSSION THREAD

**Discussion by Slavecode** [open-wa/wa-automate-nodejs#2552] [Q&A]
what is the maximum limit for sending video, audio, and pdf in MD?

**Comment by smashah** [comment]
Iirc files can go up to 100mb and videos can go up to 64mb (or the other way around)

**Reply by devmarinho** [reply]
I had the same problem, but the size of the video was 17,5 mb. See log error 

Error: Evaluation failed: Error: MediaFileTooLarge: File (video/mp4) size = 17178292
    at m.sendToChat (https://web.whatsapp.com/bootstrap_qr.b472ccf746046d1080f9.js:69:337583)
    at __puppeteer_evaluation_script__:1:189759

**Reply by smashah** [reply]
Just tested with a 18mb file and it worked:

Make sure you've set `--use-chrome` or `useChrome: true` if you want to send videos.

```bash
curl -X 'POST' \
  'http://localhost:8060/sendImage' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "args": {
    "to": "123455@c.us",
    "file": "https://file-examples.com/storage/feb8f98f1d627c0dc94b8cf/2017/04/file_example_MP4_1920_18MG.mp4",
    "filename": "sample.mp4",
    "caption": "Hello World!",
    "waitForId": true
  }
}'
```

**Reply by devmarinho** [reply]
The setting was `useChrome` already setted to true. I could send video with size 1.5mb, but I couldn't do that specifically. Try its: https://www.youtube.com/watch?v=hIS6p9SaQYM

I'm downloading it and trying to send.

**Reply by Sansekai** [reply]
Same problem

**Reply by smashah** [reply]
If you share the exact file/url as i did above then i can test it. most likely it is an encoding issue and you have to have a ffmpeg step in the middle to make sure the video is mp4

**Reply by Slavecode** [reply]
> 

![20220527_001443](https://user-images.githubusercontent.com/43001614/170530786-8c04f313-61ef-49b3-939b-b4d4b3de4ac6.jpg)
I've installed ffmpeg and tried the mp4 format with a size of 21MB.

My options :
`sessionId: 'Ajk',
    multiDevice: true,
    blockCrashLogs: true,
    disableSpins: true,
    headless: true,
    logConsole: false,
    popup: true,
    qrTimeout: 0,
    authTimeout: 60,
    restartOnCrash: start,
    cacheEnabled: false,
    useChrome: true,
    waitForRipeSession: true,
    killProcessOnBrowserClose: true,
    throwErrorOnTosBlock: false`

**Reply by smashah** [reply]
Don't being up issues in discussions.

If you have links to specific video files then share in an issue with the code so it can be tested.

**Reply by devmarinho** [reply]
That's the example video that i mentioned:

https://drive.google.com/file/d/1OY65SqB0DRlPAPAWS2jgWJXrA83KBrJt/view?usp=sharing

I have downloaded this video on mp4 and tried to send, but i get the error. 

My config: 
      blockCrashLogs: false,
      disableSpins: false,
      hostNotificationLang: 'PT_BR',
      logConsole: false,
      viewport: {
          width: 1920,
          height: 1200
      },
      popup: 3012,
      multiDevice: true,
      defaultViewport: null,
      sessionId: 'Gramont-Bot1',
      headless: headless,
      qrTimeout: 0,
      authTimeout: 60,
      restartOnCrash: start,
      cacheEnabled: true,
      useChrome: true,
      killProcessOnBrowserClose: true,
      throwErrorOnTosBlock: true,
      inDocker: false,

**Reply by Sansekai** [reply]
> That's the example video that i mentioned:
> 
> https://drive.google.com/file/d/1OY65SqB0DRlPAPAWS2jgWJXrA83KBrJt/view?usp=sharing
> 
> I have downloaded this video on mp4 and tried to send, but i get the error.
> 
> My config: blockCrashLogs: false, disableSpins: false, hostNotificationLang: 'PT_BR', logConsole: false, viewport: { width: 1920, height: 1200 }, popup: 3012, multiDevice: true, defaultViewport: null, sessionId: 'Gramont-Bot1', headless: headless, qrTimeout: 0, authTimeout: 60, restartOnCrash: start, cacheEnabled: true, useChrome: true, killProcessOnBrowserClose: true, throwErrorOnTosBlock: true, inDocker: false,

after i update my puppeteer from 14.1.0 to 14.3.0 i can send videos more than 15 mb.
![Screenshot_20220619-200002_WhatsApp](https://user-images.githubusercontent.com/34540627/174486008-441ff017-592e-4e37-94b3-7d9dbfe6ee3f.jpg)

**Reply by SamPandey001** [reply]
> Iirc files can go up to 100mb and videos can go up to 64mb (or the other way around)

Ohh Great

**Reply by devmarinho** [reply]
> > That's the example video that i mentioned:
> > https://drive.google.com/file/d/1OY65SqB0DRlPAPAWS2jgWJXrA83KBrJt/view?usp=sharing
> > I have downloaded this video on mp4 and tried to send, but i get the error.
> > My config: blockCrashLogs: false, disableSpins: false, hostNotificationLang: 'PT_BR', logConsole: false, viewport: { width: 1920, height: 1200 }, popup: 3012, multiDevice: true, defaultViewport: null, sessionId: 'Gramont-Bot1', headless: headless, qrTimeout: 0, authTimeout: 60, restartOnCrash: start, cacheEnabled: true, useChrome: true, killProcessOnBrowserClose: true, throwErrorOnTosBlock: true, inDocker: false,
> 
> after i update my puppeteer from 14.1.0 to 14.3.0 i can send videos more than 15 mb. ![Screenshot_20220619-200002_WhatsApp](https://user-images.githubusercontent.com/34540627/174486008-441ff017-592e-4e37-94b3-7d9dbfe6ee3f.jpg)

I updated my puppeteer to 14.3.0 and it made no difference. Is there any puppeter lib more? There are puppeteer-core, puppeteer-extra in project, should i update them too?

**Comment by Slavecode** [comment]
> Iirc files can go up to 100mb and videos can go up to 64mb (or the other way around)

![20220226_192810](https://user-images.githubusercontent.com/43001614/155841543-afa66c1a-a334-4b49-80b7-15739ff1f752.jpg)
I sent a video that is 20mb in size and an error like this appears

**Comment by smashah** [comment]
Please keep an eye on #2675 which will increase the theoretical throughput between the node process and the browser page context.
